### PR TITLE
Run2-hcx157 Update getGeometry method for HcalGeometry to enable standard access

### PR DESCRIPTION
**Pull request by bsunanda**

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by bsunanda** [comment]
@ianna Can you please take a look and sign this?

File: Geometry/HcalTowerAlgo/src/HcalGeometry.cc [hunk 1]
```diff
@@ -83,6 +83,27 @@ HcalGeometry::getValidDetIds( DetId::Detector det,
...
+#ifdef EDM_ML_DEBUG
+  std::cout << "HcalGeometry::getGeometry for " << HcalDetId(id) << "  " 
+	    << m_mergePosition << " ";
+#endif
+  if (!m_mergePosition) {
+#ifdef EDM_ML_DEBUG
+    std::cout << m_topology.detId2denseId(id) << " " 
+	      << cellGeomPtr(m_topology.detId2denseId(id)) << "\n";
+#endif
+    return cellGeomPtr(m_topology.detId2denseId(id));
```
Comment (davidlange6) [inline Geometry/HcalTowerAlgo/src/HcalGeometry.cc]: how about return getGeometryBase(m_topology.detId2denseId(id)) here and below.

Comment (bsunanda) [reply Geometry/HcalTowerAlgo/src/HcalGeometry.cc]: Done

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@smuzaffar @davidlange6 @slava77 The failure is not due to this PR. Please advise

**Comment by smuzaffar** [comment]
please test
error was in IB which is fixed now

**Comment by civanch** [comment]
@bsunanda , you need comment out DEBUG statement

**Comment by davidlange6** [comment]
merge

File: Geometry/HcalTowerAlgo/src/HcalGeometry.cc [hunk 2]
```diff
@@ -11,7 +11,7 @@ typedef CaloCellGeometry::Pt3D     Pt3D     ;
 typedef CaloCellGeometry::Pt3DVec  Pt3DVec  ;
 typedef CaloCellGeometry::Tr3D     Tr3D     ;
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (slava77) [inline Geometry/HcalTowerAlgo/src/HcalGeometry.cc]: is this removed already in a follow up PR?

Comment (slava77) [reply Geometry/HcalTowerAlgo/src/HcalGeometry.cc]: I see that it was actually fixed in 
4297d2489bfac27c8fc19d37d58b5264f424ecfe
was there a PR?

**Pull request merged**